File: go-dev/ipv6.go
package main

import (
    "fmt"
	"net"
)

func main() {

    var src string  = "2003:da8:2004:1000:0a09:5292:0100:3797"
    fmt.Printf("src: %s\n", src)
    ip := net.ParseIP(src)
    fmt.Printf("dst: %s\n", ip.String())
}

File: cpp-dev/network/ipv6.c

#include <netinet/ip6.h>
#include <stdio.h>

int main(int argc, char* argv[]) {

    char* src = "2003:da8:2004:1000:a09:5292:100:3797";
    char ip[16];

    scan_ip6(src, ip);

    char buf[64];
    buf[fmt_ip6(buf,ip)]=0;

    printf("fmt ipv6:%s\n", fmt_ip6);

}
